### PR TITLE
[REFACTOR][IR] alpha_equal to structural_equal

### PR DESCRIPTION
**Pull request by zhiics**
#5159 

This PR finishes the first three items of #5159:

- [X] Replace `alpha_equal/type_alaph_equal` and `assert_alpha_equal` from with `structural_equal`

- [X] Replace all `AlphaEqual(lhs, rhs)` in C++ with `StructuralEqual()(lhs, rhs)`

- [X] Remove analysis/alpha_equal.cc

A few bugs are also fixed in this PR.

A followup PR needs to remove the `AttrEqual` and `AlphaEqualHandler` from C++.

### CODE REVIEW

**Comment by zhiics** [comment]
cc @tqchen @MarisaKirisame @junrushao1994 @jroesch

File: include/tvm/ir/type_relation.h [hunk 1]
```diff
@@ -203,7 +203,6 @@ class TypeRelationNode : public TypeConstraintNode {
 
   bool SEqualReduce(const TypeRelationNode* other, SEqualReducer equal) const {
     return
-        equal(func, other->func) &&
```
Comment (zhiics) [inline include/tvm/ir/type_relation.h]: @tqchen please check here. I think we shouldn't check `func` here as it is a callback.

Comment (tqchen) [inline include/tvm/ir/type_relation.h]: Let us keep this comparison. we can fix the EnvFunc's SEqual instead by comparing the name.

File: src/relay/op/tensor/transform.cc [hunk 2]
```diff
@@ -2142,7 +2142,9 @@ Expr MakeSplit(Expr data,
 TVM_REGISTER_GLOBAL("relay.op._make.split")
 .set_body([](const TVMArgs& args, TVMRetValue* rv) {
     if (args.type_codes[1] == kDLInt) {
-      *rv = MakeSplit(args[0], tir::make_const(DataType::Int(64), int64_t(args[1])), args[2]);
+      *rv = MakeSplit(args[0],
+                      tir::make_const(DataType::Int(32), static_cast<int>(args[1])),
```
Comment (zhiics) [inline src/relay/op/tensor/transform.cc]: Without this change, all tests in `tests/python/relay/test_pass_combine_parallel_dense.py` fail

File: include/tvm/ir/type_relation.h (hunk 1 above)
Comment (tqchen) [reply include/tvm/ir/type_relation.h]: Let us keep the comparison, try to apply the env func related fix here:

https://github.com/apache/incubator-tvm/pull/5160/commits/6301f214bb0f4319e7ca4bb5a2e7b3deb0a6d441

File: src/relay/op/tensor/transform.cc (hunk 2 above)
Comment (tqchen) [reply src/relay/op/tensor/transform.cc]: can we comment about why?

Comment (tqchen) [reply src/relay/op/tensor/transform.cc]: Is it because the transformed pass uses int32 to construct the attributes?

Comment (zhiics) [reply src/relay/op/tensor/transform.cc]: Yeah, the transformed graph uses int32. The other way is probably to find out where it is transformed and use int64 there.

Comment (tqchen) [reply src/relay/op/tensor/transform.cc]: we can go with this fix for now and check back later

**Comment by tqchen** [comment]
@zhiics please rebase and check the testcases

**Comment by zhiics** [comment]
@tqchen sure. Seems a few tests are broken. I will look into after base.

**Comment by tqchen** [comment]
@zhiics can you comment a bit about what test-cases hit the incomplete type handling? That part might need some extra thoughts, I can merge this PR for now and we follow up on the particular item. cc @jroesch @MarisaKirisame

**Comment by zhiics** [comment]
@tqchen sure. It is this one:

https://github.com/apache/incubator-tvm/blob/919ae889638555b82de2d124d5f3e08d76bf789b/tests/python/relay/test_ir_structural_equal_hash.py#L66

**Comment by tqchen** [comment]
I see, seems it makes sense now then. Thanks @zhiics !

**Pull request merged**